operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/retry/GenericRetryExecution.java

```diff
@@ -1,12 +1,13 @@
 package io.javaoperatorsdk.operator.processing.retry;
 
 import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
 
 public class GenericRetryExecution implements RetryExecution {
 
   private final GenericRetry genericRetry;
 
-  private volatile int lastAttemptIndex = 0;
+  private AtomicInteger lastAttemptIndex = new AtomicInteger(0);
```

Comment:
Do we actually have multiple threads accessing that state concurrently?

Reply:
checked and atctually don't now, good catch! Changhed it

Reply:
Did you push the changes, I don't see them?

Reply:
now I did :)